src/mlj_model_interface.jl

```diff
@@ -142,46 +132,92 @@ function MLJModelInterface.fit(model::MLJFluxModel,
         deepcopy(rng),
         move,
     )
-    fitresult = MLJFlux.fitresult(model, Flux.cpu(chain), y)
 
-    report = (training_losses=history, )
+    # Extract embedding matrices
+    enable_entity_embs &&
+        (embedding_matrices = get_embedding_matrices(chain, cat_inds, featnames))
+
+    # Prepare fitresult 
+    fitresult_args = (model, Flux.cpu(chain), y)
+
+    # Prepare report
+    report = (training_losses = history,)
+
+    # Modify cache and fitresult if entity embeddings enabled
+    if enable_entity_embs
+        cache = (cache..., entityprops, entityemb_output_dim, ordinal_mappings, featnames)
+        fitresult =
+            MLJFlux.fitresult(fitresult_args..., ordinal_mappings, embedding_matrices)
+    else
+        fitresult = MLJFlux.fitresult(fitresult_args...,)
+    end
 
```

Comment:
I wonder if it might be cleaner to dispense with some of the case distinctions  when constructing `fitresult` and `cache`. Rather, just return empty objects instead of *no* objects. For example, `get_embedding_matrices(chain, cat_inds, featnames)` probably makes sense already for empty `cat_inds`, returning an empty list of matrices, which is fine.

I agree it makes sense to drop the embedding layer when its trivial because that is a performance-critical element called on thousands of times, but otherwise all this extra if-then-else logic makes it harder to follow the code, I find. And it kind of bothers me that the form of `fitresult` and `cache` depend on the data. A maintainer of the code will not expect this. 

In my experience case-distinctions are the source of a lot of bugs, cognitive dissonance, and code complexity and should be avoided where it's natural to do so. And they add a testing burden as well. 

You've done a marvelous job to sort all this out, but I'm anticipating the pain of the added code complexity created for future maintenance and extension.

Reply:
> I wonder if it might be cleaner to dispense with some of the case distinctions when constructing fitresult and cache.

I share your sentiment @ablaom  but the cost will be that the `ImageClassifier` will take these extra arguments and do nothing with them.

My impression is that it's likely and beneficial for developers in the future to contribute with more advanced but frequently used neural models such as RNNs, LSTMs, Transformers or autoencoders. In all of these cases and others, the entity embedding stuff is quite irrelevant.

If we stop making the case distinction, the `fitresult` for all such futuristic models including the currently existing `ImageClassifier` will change to accomodate the extra inputs but do nothing with them. 

It seems to me that a reader of the code would have familiarized themselves with the distinction by the time they see that because it's used in critical points before this so I don't see highly significant value from removing it here.

If you like, we can add just a little more redundancy and have two fits and two updates one for each case (entity_embeddings_enabled or not).

Reply:
Alright. I agree this is cleaner but not sure if its less confusing for developers of new models that don't need entity embeddings. Done in f60f789acb0b02452565a4654619f133be6fd60b